transport/src/main/java/io/scalecube/transport/TransportImpl.java

```diff
@@ -84,29 +89,27 @@ public TransportImpl(TransportConfig config) {
   }
 
   private Mono<Transport> bind0(int port) {
-    return Mono.defer(
-        () ->
-            toMono(
-                    bootstrapFactory
-                        .serverBootstrap()
-                        .childHandler(incomingChannelInitializer)
-                        .bind(port))
-                .doOnSuccess(
-                    channel -> {
-                      serverChannel = (ServerChannel) channel;
-                      address = toAddress(serverChannel.localAddress());
-                      networkEmulator = new NetworkEmulator(address, config.isUseNetworkEmulator());
-                      networkEmulatorHandler =
-                          config.isUseNetworkEmulator()
-                              ? new NetworkEmulatorHandler(networkEmulator)
-                              : null;
-                      LOGGER.info("Bound cluster transport on: {}", address);
-                    })
-                .doOnError(
-                    cause ->
-                        LOGGER.error(
-                            "Failed to bind cluster transport on port={}, cause: {}", port, cause))
-                .thenReturn(TransportImpl.this));
+    return TcpServer.create()
+        .runOn(loopResources)
+        .addressSupplier(() -> new InetSocketAddress(config.getPort()))
+        .bootstrap(b -> BootstrapHandlers.updateConfiguration(b, "inbound", incomingPipeline))
+        .handle((inb, outb) -> inb.receive().aggregate().retain().then())
+        .bind()
+        .doOnSuccess(
+            s -> {
+              this.server = s;
```

Comment:
Plz create private method `onBind()`  and move logic on lines 101 - 107 there. I always wanted to do that : |

Reply:
+